reference/6/Microsoft.PowerShell.Core/Remove-Module.md

```diff
@@ -205,6 +205,20 @@ You can pipe module names and module objects to **Remove-Module**.
 This cmdlet does not generate any output.
 
 ## NOTES
+When removing a module, there is an event on the module that will execute. This allows a module to react to being removed and maybe free up resources. Example:
+
+$PSF_OnRemoveScript = {
```

Comment:
I realize this example comes from your PSFramework module, but I think it can be a bit confusing to new users and would suggest making it more generic:

```powershell
$OnRemoveScript = {
  # perform cleanup
  $cachedSessions | Remove-PSSession
}

$ExecutionContext.SessionState.Module.OnRemove += $OnRemoveScript
```

Reply:
Absolutely, thanks for pointing this out - a bit of mental laziness crept up on me there. Updated, I hope the linebreaks are ok in that scriptblock, given the limitations of platyPS.